crates/rpc/src/v1/types/transaction_filter.rs

```diff
@@ -0,0 +1,75 @@
+use ethereum_types::{Address, U256};
+use v1::types::Transaction;
+
+#[allow(non_camel_case_types)]
+#[derive(Debug, Deserialize, Serialize)]
+#[serde()]
+pub enum EqFilterArgument<T: Eq> {
+    eq(T),
+    Nil,
+}
+
+impl<T: Eq> Default for EqFilterArgument<T> {
+    fn default() -> Self {
+        Self::Nil
+    }
+}
+
+impl<T: Eq> EqFilterArgument<T> {
+    fn matches(&self, value: &T) -> bool {
+        match self {
+            Self::eq(expected) => value == expected,
+            Self::Nil => true,
+        }
+    }
+}
+
+#[allow(non_camel_case_types)]
+#[derive(Debug, Deserialize, Serialize)]
+#[serde()]
+pub enum ValueFilterArgument {
+    eq(U256),
+    lt(U256),
+    gt(U256),
+    Nil,
+}
+
+impl Default for ValueFilterArgument {
+    fn default() -> Self {
+        Self::Nil
+    }
+}
+
+impl ValueFilterArgument {
+    fn matches(&self, value: &U256) -> bool {
+        match self {
+            ValueFilterArgument::eq(expected) => value == expected,
+            ValueFilterArgument::lt(threshold) => value < threshold,
+            ValueFilterArgument::gt(threshold) => value > threshold,
+            ValueFilterArgument::Nil => true,
+        }
+    }
+}
+
+#[allow(non_camel_case_types)]
```

Comment:
camel case is used for json rpc

Reply:
I see that original pull request used `gas_price` but all other jsonrpc filed are camel case and in this case I think it is better to be consistent and use `gasPrice`